src/components/EditCard/ViewActions.vue

```diff
@@ -0,0 +1,63 @@
+<template>
+  <div>
+    <v-menu v-for="action in actions" offset-y :key="action.emit">
+      <template v-slot:activator="{ on: menu }">
+        <v-tooltip bottom>
+          <template v-slot:activator="{ on: tooltip }">
+            <v-btn icon v-on="{ ...tooltip, ...menu }" v-if="action.menu">
```

Comment:
What do the spread operators do here?

Reply:
This is a great question! I actually don't really know the exact answer. However, after doing some research, here are two insightful threads:
- https://github.com/vuetifyjs/vuetify/issues/6866#issuecomment-499709110
- https://stackoverflow.com/questions/55188478/meaning-of-v-slotactivator-on

I don't fully understand it, but I'm going to try to put it in my own words: 

When we call
```html
<template v-slot:activator="{ on }">
```
we're actually passing in an object that represents the scope of the parent `v-menu` (in this case). The scope object has an `on` property that we are referencing using [object destructuring](https://babeljs.io/docs/en/learn#destructuring). The `on` property is itself an object which has a bunch of information for [handling events](https://vuejs.org/v2/guide/events.html). We use it by passing it to `v-on`.

However, in this case, we have multiple activators and scopes, so we have to name the `on` objects different things. I named them according to the parent, so `menu` and `tooltip`. 

Now remember, these are objects that describe how to handle events (like hovering or clicking). We need to combine that information into one object that the `v-btn` can use when handling events. Therefore, we use the spread operators as a nice way of combining the two objects.

The simplified picture I have in my mind is:
```typescript
// how to handle events for menu, like clicking to show the menu
menu = { ... , click: showMenu, ... } 
// how to handle events for tooltip, probably just hover
tooltip = { ... , hover: showTooltip, ... }
// combines the effects ==> { ... , click: showMenu, ... , hover: showTooltip, ... } 
menuAndTooltip = { ...menu, ...tooltip }  
// now the button can use `menuAndTooltip` and know what to do for both hover and click
```

Hopefully this is a correct understanding, and hopefully this helps! Definitely check out the links if you want to verify / understand more.

Reply:
Wow this is a solid explanation, thanks! Your understanding makes sense to me but I will take a look at those links to see if I can learn more